jobs/src/main/scala/dcos/metronome/model/JobRunSpec.scala

```diff
@@ -58,17 +58,19 @@ object JobRunSpec {
       val envVarDefinedSecretNames = jobRunSpec.env.values.collect { case EnvVarSecret(secretName) => secretName }.toSet
       val providedSecretNames = jobRunSpec.secrets.keySet
 
-      check(jobRunSpec.cmd.isDefined || jobRunSpec.docker.exists(d => d.image.nonEmpty), JobRunSpecMessages.cmdOrDockerValidation)
+      check(jobRunSpec.cmd.isDefined || jobRunSpec.docker.exists(d => d.image.nonEmpty) || jobRunSpec.ucr.nonEmpty, JobRunSpecMessages.cmdOrDockerValidation)
       check(envVarDefinedSecretNames == providedSecretNames, JobRunSpecMessages.secretsValidation(envVarDefinedSecretNames, providedSecretNames))
+      check(!(jobRunSpec.docker.nonEmpty & jobRunSpec.ucr.nonEmpty), JobRunSpecMessages.onlyDockerOrUcr)
 
       violations.headOption.getOrElse(Success)
     }
   }
 }
 
 object JobRunSpecMessages {
-  val cmdOrDockerValidation = "Cmd or docker image must be specified"
+  val cmdOrDockerValidation = "Cmd, docker or ucr image must be specified"
```

Comment:
we should have consistency with line 75 below... Docker (not docker) and UCR.   Also this message could be more clear IMO.